dags/multipod/maxtext_v4_configs_aot_hybridsim.py

```diff
@@ -0,0 +1,109 @@
+SCHEDULED_TIME = "0 10 * * *" if composer_env.is_prod_env() else None
+
+with models.DAG(
+    dag_id="maxtext_v4_configs_aot_hybridsim",
+    schedule=SCHEDULED_TIME,
+    tags=["multipod_team", "maxtext", "stable", "nightly"],
+    start_date=datetime.datetime(2024, 2, 19),
+    catchup=False,
+    concurrency=10,
+) as dag:
+  # Test setup values
+  model_configs = {
+      # accelerator: [(model_size, num_cores), ...],
+      TpuVersion.V4: [("22b", 128), ("52b", 384)],
+  }
+  num_slices = [1, 2]
+  clusters = {
+      # accelerator: [(cluster_name, cluster_zone, cluster_project, num_cores)],
+      TpuVersion.V4: (
+          ClusterName.V4_8_MULTISLICE_CLUSTER.value,
+          Zone.US_CENTRAL2_B.value,
+          Project.TPU_PROD_ENV_MULTIPOD.value,
+          8,
+      ),
+  }
+
+  for tpu, models in model_configs.items():
+    for model_size, num_cores in models:
+      for n in num_slices:
+        test_group_id = (
+            f"{model_size}-{n}xv{tpu.value}-{num_cores}-aot-hybridsim"
+        )
+        with TaskGroup(group_id=test_group_id, prefix_group_id=False) as group:
+          gcs_subfolder = f"{test_owner.Team.MULTIPOD.value}/maxtext"
+          shared_gcs_location = (
+              name_format.generate_gcs_folder_location.override(
+                  task_id=f"{test_group_id}_generate_gcs_folder_location"
+              )(
+                  f"{gcs_subfolder}/maxtext_configs_aot_hybridsim/v{tpu.value}",
+                  test_group_id,
+              )
+          )
+
+          # Run AOT workload: generate HLO, upload to GCS
+          aot_cmd = (
+              'export XLA_FLAGS="--xla_dump_to=/tmp/xla_dump/"',
+              f"bash MaxText/configs/v{tpu.value}/{model_size}.sh EXECUTABLE=train_compile.py M_COMPILE_TOPOLOGY=v{tpu.value}-{num_cores} M_COMPILE_TOPOLOGY_NUM_SLICES={n}",
+              "gsutil cp gs://cloud-hybridsim-prod/desanitize_and_upload_hlo.sh .",
+              "bash desanitize_and_upload_hlo.sh LOCAL_DIR=/tmp/xla_dump/ GCS_OUTPUT_PATH=${GCS_OUTPUT}",
+          )
+          maxtext_aot = gke_config.get_gke_config(
+              tpu_version=TpuVersion.V4,
+              tpu_cores=8,
+              tpu_zone=Zone.US_CENTRAL2_B.value,
+              time_out_in_min=240,
+              test_name=f"maxtext-{model_size}-{n}xv{tpu.value}-{num_cores}-aot",
+              run_model_cmds=aot_cmd,
+              docker_image=DockerImage.MAXTEXT_TPU_JAX_STABLE.value,
+              test_owner=test_owner.RAYMOND_Z,
+          ).run(gcs_location=shared_gcs_location)
+
+          # Run HybridSim workload: read HLO from GCS, generate estimated step time
+          cluster_name, zone, project_name, cores = clusters[tpu]
+          chip_config = "default" if tpu == TpuVersion.V5E else "megacore"
+          hybridsim_cmd = (
+              "gsutil cp gs://cloud-hybridsim-prod/run_hybridsim.sh .",
+              f"bash run_hybridsim.sh GCS_PATH=${{GCS_OUTPUT}}xla_dump CHIP_CONFIG={chip_config}",
```

Comment:
How are we processing the resulting step time estimation?

Reply:
This is still in progress. The step time estimation is printed to logs and we will need to capture it and write it to a jsonlines file for XLML to parse.